### ISSUE TITLE
ArrayIndexOutOfBoundsException when rendering PostgreSQL plain SQL template containing ?@

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13489]
This code:

```java
println(DSL.using(SQLDialect.POSTGRES).render(DSL.field("?@", DSL.one())));
```

Produces an `ArrayIndexOutOfBoundsException`:

```
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 2 out of bounds for length 2
	at org.jooq.impl.Tools.renderAndBind(Tools.java:2995)
	at org.jooq.impl.SQLImpl.accept(SQLImpl.java:78)
	at org.jooq.impl.DefaultRenderContext.visit0(DefaultRenderContext.java:739)
	at org.jooq.impl.AbstractContext.visit(AbstractContext.java:307)
	at org.jooq.impl.SQLField.accept(SQLField.java:70)
	at org.jooq.impl.DefaultRenderContext.visit0(DefaultRenderContext.java:739)
	at org.jooq.impl.AbstractContext.visit(AbstractContext.java:307)
	at org.jooq.impl.DefaultDSLContext.render(DefaultDSLContext.java:709)
	at org.jooq.testscripts.JDBC.main(JDBC.java:30)
```

**Comment by lukaseder** [comment]
The problem is that the operator is the last thing in the string. Quite the edge case, but could happen e.g. when templates are combined from other templates.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.7 (https://github.com/jOOQ/jOOQ/issues/13490), and 3.15.11 (https://github.com/jOOQ/jOOQ/issues/13491)
